### PR TITLE
Feature/change pid record fields to enums

### PR DESCRIPTION
**Pull request by southeo**
The following (user-inputted fields) have been changed to enums:

- DigitalOrPhysical: ["digital", "physical"]
- ObjectType: ["handle", "doi", "digitalSpecimen", "digitalSpecimenBotany", "mediaObject", "tombstone"]
- PhysicalIdType: ["global", "combined"]
- PidStatus: ["ALL", "ACTIVE", "ARCHIVED", "DRAFT", "RESERVED", "TEST", "TEST2"]
- PreservedOrLiving: ["preserved", living"]

JsonSchemaValidator has been updated to limit the controlled terms.   

https://naturalis.atlassian.net/browse/DD-242?atlOrigin=eyJpIjoiMDc2NzFkZTUwNDFiNGQ0Nzg4NjQ0NjU5ZWMzOTA0YjAiLCJwIjoiaiJ9

### CODE REVIEW

**Comment by wouteraddink** [comment]
Physical ID Type enum is not as discussed?

Op wo 8 mrt. 2023 11:26 schreef southeo ***@***.***>:

File: src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java [hunk 1]
```diff
@@ -62,18 +62,14 @@ public ResponseEntity<String> hello() {
   public ResponseEntity<List<String>> getAllHandlesByPidStatus(
       @RequestParam(value = "pageNum", defaultValue = "0") int pageNum,
       @RequestParam(value = "pageSize", defaultValue = "100") int pageSize,
-      @RequestParam(name = "pidStatus", defaultValue = "ALL") String pidStatus)
-      throws PidResolutionException, InvalidRequestException {
+      @RequestParam(name = "pidStatus", defaultValue = "ALL") PidStatus pidStatus)
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java:65]: Can defaultValue be PidStatus.ALL instead of the String? (Not sure it is possible but would be nice)

File: src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java [hunk 2]
```diff
@@ -62,18 +62,14 @@ public ResponseEntity<String> hello() {
...
-      throw new InvalidRequestException(
-          "Invalid Input. Pid Status not recognized. Available Pid Statuses: " + VALID_PID_STATUS);
-    }
     List<String> handleList;
-    if (pidStatus.equals("ALL")) {
+    if (pidStatus.toString().equals("ALL")) {
       handleList = service.getHandlesPaged(pageNum, pageSize);
     } else {
-      handleList = service.getHandlesPaged(pageNum, pageSize, pidStatus);
+      handleList = service.getHandlesPaged(pageNum, pageSize, pidStatus.toString());
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java:72]: Why pass the string?

File: src/main/java/eu/dissco/core/handlemanager/domain/PidRecords.java [hunk 3]
```diff
@@ -86,12 +77,13 @@ public class PidRecords {
       Map.entry(DIGITAL_OBJECT_SUBTYPE, 4), Map.entry(LOC, 5), Map.entry(ISSUE_DATE, 6),
       Map.entry(ISSUE_NUMBER, 7), Map.entry(PID_STATUS, 8), Map.entry(TOMBSTONE_TEXT, 9),
       Map.entry(TOMBSTONE_PIDS, 10), Map.entry(PID_KERNEL_METADATA_LICENSE, 11),
-      Map.entry(REFERENT_DOI_NAME, 12), Map.entry(REFERENT, 13), Map.entry(DIGITAL_OR_PHYSICAL, 14),
+      Map.entry(REFERENT_DOI_NAME, 12), Map.entry(REFERENT, 13),
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/handlemanager/domain/PidRecords.java:80]: Why these changes? MEDIA_URL moved from 15 to 16?

File: src/test/java/eu/dissco/core/handlemanager/service/HandleServiceTest.java [hunk 4]
```diff
@@ -615,7 +615,7 @@ void testGetHandlesPaged() {
 
   // Exceptions
 
-  @Test
+
```
Comment (samleeflang) [inline src/test/java/eu/dissco/core/handlemanager/service/HandleServiceTest.java:618]: Is this correct the `@Test` seems to be removed?

File: src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java (hunk 1 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java:65]: It can't be, it has to be a string. :/ 
Tried `PidStatus.valueOf("ALL")` but I got "Attribute value must be consistent" from the IDE.

File: src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java (hunk 2 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java:72]: The associated repository function:

```
 return context
        .selectDistinct(HANDLES.HANDLE)
        .from(HANDLES)
        .where(HANDLES.TYPE.eq(PID_STATUS.getBytes(StandardCharsets.UTF_8)))
        .and(**HANDLES.DATA.eq(pidStatus))**
```

My thinking is we need to transform this value eventually so the DB can match the value, might as well do it here.

File: src/main/java/eu/dissco/core/handlemanager/domain/PidRecords.java (hunk 3 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/domain/PidRecords.java:80]: There was a key conflict with specimenHost, which is also at index 15. 
 
Updated datamodel to reflect this:
https://docs.google.com/document/d/1eD1cPkk0bIzbuezwbVj89DG5AiCMyWHFQkhkik9A7M4/edit#heading=h.ghnb1aaw0ji2

File: src/test/java/eu/dissco/core/handlemanager/service/HandleServiceTest.java (hunk 4 above)
Comment (southeo) [reply src/test/java/eu/dissco/core/handlemanager/service/HandleServiceTest.java:618]: Those tests are obsolete; they've been removed

File: src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java (hunk 2 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java:72]: This has been changed to `pidStatus.getBytes()` as that's what the repository actually needs

**Review by samleeflang — APPROVED** [review]
